clients/application/src/main/java/org/eclipse/hono/client/application/MessageConsumer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client.application;
+
+import io.vertx.core.Future;
+
+/**
+ * A client that consumes messages from Hono's northbound APIs.
+ *
+ * @param <T> The type of messages consumed by this client.
+ */
+public interface MessageConsumer<T extends Message> {
```

Comment:
`<T extends DownstreamMessage>`, once `DownstreamMessage` is an interface?

Reply:
I would use this for command responses as well, which are not conform to the `DownstreamMessage` interface.